File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-1.go
func largestRectangleArea(heights []int) int {
	maxArea := 0
	for i := 0; i < len(heights); i++ {
		left, right := -1, -1
		area := heights[i]
		if i > 0 {
			left = i - 1
		} 
		if i + 1 < len(heights) {
			right = i + 1
		}

		for left >= 0 && heights[left] >= heights[i] {
			area += heights[i]
			left--
		} 
		for right < len(heights) && right > 0 && heights[right] >= heights[i] {
			area += heights[i]
			right++
		}

		maxArea = max(maxArea, area)
	}

	return maxArea
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left, right, maxprof = 0, 1, 0
        while right < len(prices): 
            diff = prices[right] - prices[left]
            if diff < 0:
                left = right
            else:
                maxprof = max(maxprof, diff)
            right += 1
        return maxprof 